docs_src/explanations/how-scaph-computes-per-process-power-consumption.md

```diff
@@ -1,7 +1,53 @@
+
+This is easy to understand, but the reality is somewhat different.
+
+![what scaphandre does - per process power usage](../img/what-scaphandre-does.png)
+
+#### Timesharing of work
+
+As we know computers work on lots of different jobs at the same time - they work on one job for short interval of time, then another, and another and so one. You'll often see these [small intervals of time referred to as [_jiffies_][].
+
+[jiffies]: https://www.anshulpatel.in/post/linux_cpu_percentage/
+
+![work on jobs is split into jiffies](../img/jiffies.png)
+
+In a given amount of time, certain jobs that are more important, or resource intensive will use more jiffies than others. Fortunately, each job keeps a running total of the total jiffies allocated to it, so if we know how many jiffies have been used in total, it can give us an idea how much of a machine's resources are being used by a given process.
+
+![work on jobs is split into jiffies](../img/total-time-share.png)
+### Going from share of resources to actual power figures
+
+It's possible without Scaphandre to understand how large a share of a machines' resources are being used by a given process, but if we want to understand the power used per process, we need to know how much power is being used by the machine in absolute terms.
 
+To do this, we need a sensor of some kind to track power usage by the machien itself. Some servers have these like with Intel's RAPL sensots,, making it possible to understand how much power is being used by CPUs, GPUs and so on.
+
+![Sensors provide power over time](../img/power-over-time.png)
+
+If we combine all the intervals when our job is worked on, _and_ we know how much power is being drawn at those moments in time, we're in a better place.
+
+We get an idea of how much power is being used by our job in total, in absolute terms, by looking at the power draw during all the moments our particular job is being worked on - that is, the power drawn for 'our' jiffies.
+
+![Combined we can see how much the power during 'our' jiffies](../img/power-and-share-of-usage.png)
+
+Finally once we have these, if group them together and add together the power used when 'our' jiffies are worked on, we can arrive at a useable figure.
+
+![Combined we can see how much the power during 'our' jiffies](../img/power-by-process.png)
+
+As long are able to read from sensors that can share how much power is being used by a server at a given moment, and how much of the time  being allocated to our processes during those moments, we can calculate these figures.
+
+This can work with virtualised environments, as long the guest virtual machine or guest container has access to readings provided by the host physical machine. Scaphandre supports exposing power info to virtual machines run using QEMU/KVM. (More on scaphandre in virtualised scenarios to come.).
```

Comment:
We should maybe mark the difference between virtual machines and containers, as it is not the same challenge.

Reply:
I think I'll need a bit more input in explaining the different challenges here, as I don't think I understand the container side problems well enough to write about them in detail yet.

This is partly why i was using less specific terms like _virtualised environments_, or _host physical machine_.

I can make an issue to document the different challenges faced in when working with different forms of virtualisation. 👍 ?

Reply:
If you want to, that would be great !